lib/view/select_address.dart

```diff
@@ -1,101 +1,184 @@
+                      ),
+                    ),
+                  ] else
+                    ...widget.addresses.map((e) => AddressView(
+                          key: Key(e.address),
+                          address: e,
+                          selectedAddress: _selectedAddress,
+                          onTap: () {
+                            setState(() {
+                              _selectedAddress = e.address;
+                            });
+                          },
+                          onDoneLoading: (notShowing) {
+                            setState(() {
+                              _isViewing[e.address] = !notShowing;
+                            });
+                          },
+                          minimumCryptoBalance: minimumCryptoBalance,
+                        ))
+                ],
+              )),
             ),
-          )
-        ],
-      ));
+            Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 14),
+              child: Row(
+                children: [
+                  Expanded(
+                      child: Column(
+                    children: [
+                      PrimaryButton(
+                          text: widget.selectButton ?? 'connect'.tr(),
+                          enabled: _selectedAddress != null,
+                          onTap: () {
+                            Navigator.pop(context, _selectedAddress);
+                          }),
+                      const SizedBox(
+                        height: 10,
+                      ),
+                      OutlineButton(
+                          text: 'cancel'.tr(),
+                          onTap: () {
+                            Navigator.pop(context);
+                          })
+                    ],
+                  )),
+                ],
+              ),
+            )
+          ],
+        ));
+  }
+
+  bool _noAddressHasEnoughBalance() =>
+      widget.addresses.every((element) => _isViewing[element.address] == false);
 }
 
-class AddressView extends StatelessWidget {
+class AddressView extends StatefulWidget {
   const AddressView({
     required this.address,
     super.key,
     this.selectedAddress,
     this.onTap,
+    this.minimumCryptoBalance = 0,
+    this.onDoneLoading,
   });
 
   final WalletAddress address;
   final String? selectedAddress;
   final Function()? onTap;
+  final int minimumCryptoBalance;
+  final Function(bool notShowing)? onDoneLoading;
+
+  @override
+  State<AddressView> createState() => _AddressViewState();
+}
+
+class _AddressViewState extends State<AddressView> {
+  Pair<String, String>? _balance;
+  bool _didLoad = false;
+
+  @override
+  void initState() {
+    super.initState();
+    unawaited(_loadBalance());
+  }
+
+  Future<void> _loadBalance() async {
+    final cryptoType = CryptoType.fromSource(widget.address.cryptoType);
+    _balance = await getAddressBalance(
```

Comment:
this func should not have `minimumCryptoBalance ` as param. It just should get and return the cryto balance without any check.